scanners/zap/zap_local.py

```diff
@@ -0,0 +1,172 @@
+        result = subprocess.run(["zap.sh", "-cmd", "-addonupdate"], check=False)
+        if result.returncode != 0:
+            logging.warning(
+                f"The ZAP addon update process did not finish correctly, and exited with code {result.returncode}"
+            )
+
+        # Now the real run
+        logging.info(f"Running ZAP with the following command:\n{self.zap_cli}")
+        result = subprocess.run(self.zap_cli, check=False)
+        logging.debug(
+            f"ZAP returned the following:\n=====\n{pp.pformat(result)}\n====="
+        )
+
+        # Zap's return codes : https://www.zaproxy.org/docs/desktop/addons/automation-framework/
+        if result.returncode in [0, 2]:
+            # 0: ZAP returned correctly. 2: ZAP returned warning
+            logging.info(
+                f"The ZAP process finished with no errors, and exited with code {result.returncode}"
+            )
+            self.state = State.DONE
+        else:
+            # 1: Zap hit an error
+            logging.warning(
+                f"The ZAP process did not finish correctly, and exited with code {result.returncode}"
+            )
+            self.state = State.ERROR
+
+    def postprocess(self):
+        logging.info("Running postprocess for the ZAP Host environment")
+        if not self.state == State.DONE:
+            raise RuntimeError(
+                "No post-processing as ZAP has not successfully run yet."
+            )
+
+        super().postprocess()
+
+        if not self.state == State.ERROR:
+            self.state = State.PROCESSED
+
+    def cleanup(self):
+        logging.info("Running cleanup for the ZAP Host environment")
+
+        if not self.state == State.PROCESSED:
+            raise RuntimeError("No cleanning up as ZAP did not processed results.")
+
+        logging.debug(f"Deleting temp directory {self._host_work_dir()}")
+        shutil.rmtree(self._host_work_dir())
+
+        super().cleanup()
+
+        if not self.state == State.ERROR:
+            self.state = State.CLEANEDUP
+
+    ###############################################################
+    # PROTECTED METHODS                                           #
+    # Accessed by Zap parent only                                 #
+    # + MUST be implemented                                       #
+    ###############################################################
+
+    def _add_env(self, key, value=None):
+        """Environment variable to be added to the container.
+        If value is None, then the value should be taken from the current host
+
+        In "localhost" mode, simply add the environment in the python process
+        It will be copied over to ZAP.
+        If `value` is None, then do nothing, as it means it's already set in
+        python's environment
+        """
+        if value is not None:
+            os.environ[key] = value
+
+    ###############################################################
```

Comment:
this comment block should be deleted apparently

Reply:
done